distributednetworkmanager/src/main/java/com/gruppo1/distributednetworkmanager/KadAction.java

```diff
@@ -226,24 +236,41 @@ public boolean isValid() {
         return true;
     }
 
+    /**
+     * Check if payload's content type matches with payloadType passed as parameter.
+     *
+     * @param payloadType The type of the payload.
+     * @param payload The string which contains the value of the payload.
+     * @return True if the payload string contains a value of the correct type,false otherwise.
+     */
     public static boolean payloadMatchesType(PayloadType payloadType, String payload) {
         switch (payloadType) {
             case IGNORED:
-                return true;
             case BOOLEAN:
-                Boolean.valueOf(payload);
+                return true;
             case NODE_ID:
-                //TODO
+                BitSet bitset=BitSetUtils.decodeHexString(payload);
+                return bitset.length()==ID_LENGTH;
             case PEER_ADDRESS:
                 return SMSPeer.isAddressValid(payload) == SMSPeer.PhoneNumberValidity.ADDRESS_VALID;
             case RESOURCE:
-                //metodo statico parse resource
+                try {
+                    return StringResource.parseString(payload, SEPARATOR).isValid();
```

Comment:
Wait, are we using the same separator for the parameters AND the resource name:value?
the line `String[] parameteres = messageBody.split(SEPARATOR);` will fail

Reply:
You're right I should change that SEPARATOR with the separatore used on resources

Reply:
My original idea was to not even use a separator, since everything is a constant (we know that the number of parts can't go over k and we can define a constant min and max value for operation id, like from 0 to 99). I'd choose this second option instead of using a different separator, we don't know for sure the user won't use our same separator somewhere.
If this solution is used then 0 must be used to fill any voids (i.e. operation id or max parts should be 0X instead of X) and to get the String array with the various parts, substrings should be used

Reply:
As for StringResource, I made a mistake allowing the method to take a separator, such separator should be known by the Resource class itself.

Reply:
Well, letting the user define the separator would allow avoiding certain problems (but of course one must be careful and things like this can happen)